File: replace_theme.py
import os
import re

directory = r"c:\Users\Agaru\OneDrive\Documents\visual studio 1\PADHEGAA\my-first-website\src\app"

# We must be careful to match rgba(139, 92, 246, ...) and other variations
# #8b5cf6 -> #305229
# 139, 92, 246 -> 48, 82, 41
# #ec4899 -> #4a7c3f
# 236, 72, 153 -> 74, 124, 63

replacements = [
    (re.compile(r"#8b5cf6", flags=re.IGNORECASE), "#305229"),
    (re.compile(r"139,\s*92,\s*246", flags=re.IGNORECASE), "48, 82, 41"),
    (re.compile(r"#ec4899", flags=re.IGNORECASE), "#4a7c3f"),
    (re.compile(r"236,\s*72,\s*153", flags=re.IGNORECASE), "74, 124, 63")
]

count = 0
for root, _, files in os.walk(directory):
    for file in files:
        if file.endswith(".css") or file.endswith(".tsx"):
            path = os.path.join(root, file)
            with open(path, "r", encoding="utf-8") as f:
                content = f.read()
            
            orig = content
            for regex, rep in replacements:
                content = regex.sub(rep, content)
                
            if content != orig:
                with open(path, "w", encoding="utf-8") as f:
                    f.write(content)
                count += 1
                print(f"Updated {path}")

print(f"Total files updated: {count}")

File: replace_theme.js
const fs = require('fs');
const path = require('path');

const directory = path.join(__dirname, 'src', 'app');

function walkSync(dir) {
    const files = fs.readdirSync(dir);
    for (const file of files) {
        const filePath = path.join(dir, file);
        const stat = fs.statSync(filePath);
        if (stat.isDirectory()) {
            walkSync(filePath);
        } else if (filePath.endsWith('.css') || filePath.endsWith('.tsx') || filePath.endsWith('.js') || filePath.endsWith('.ts')) {
            let content = fs.readFileSync(filePath, 'utf8');
            let orig = content;

            // Handle Hex
            content = content.replace(/#8b5cf6/gi, '#305229');
            content = content.replace(/#ec4899/gi, '#4a7c3f');

            // Handle RGB with any spacing (use global and case-insensitive)
            content = content.replace(/139,\s*92,\s*246/gi, '48, 82, 41');
            content = content.replace(/236,\s*72,\s*153/gi, '74, 124, 63');

            if (content !== orig) {
                fs.writeFileSync(filePath, content, 'utf8');
                console.log(`Updated: ${filePath}`);
            }
        }
    }
}

walkSync(directory);
console.log("Finished script.");
